ocaml/typing/mode.ml

```diff
@@ -1259,6 +1259,7 @@ module Comonadic_with_regionality = struct
   end
 
   include Common (Obj)
+  open Obj
```

Comment:
If we're going to `open Obj`, then best to get rid of the various `Obj.` uses below.

Reply:
Yeah, I was uncertain about this: this would introduce changes unrelated to this PR; those changes can be done in the future lazily.

Reply:
During a meeting today we briefly discussed this, and think it's fine to leave it as is.